### PR TITLE
Introduce tj/migrate

### PR DESCRIPTION
**Pull request by otofune**
- [x] Add tj/migrate support
  * [x] Support MongoDB as persistent backend
- [x] Add migration for changes at https://github.com/crowi/crowi/commit/c7b78c418d06364de40704a9943b713ad5dae1fe

### CODE REVIEW

File: migrate-mongo-store.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      },
+      at: {
+        type: Date,
+        required: true,
+      },
+    },
+  ],
+})
+migrationSchema.methods.toEntry = function() {
+  const m = this
```
Comment (Sw0rdstream) [inline migrate-mongo-store.js]: unnecessary `this`

File: migrate-mongo-store.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const entry = {
+    title: m.title,
+    timestamp: m.status === 'up' ? m.at.getTime() : null,
+  }
+  return entry
```
Comment (Sw0rdstream) [inline migrate-mongo-store.js]: literal variable `entry` is not accessed as left value. So the declaration is not needed.

just
```
return {
    title: this.title,
    timestamp: this.status === 'up' ? this.at.getTime() : null,
  }
```

File: package-lock.json [hunk 3]
```diff
@@ -4373,7 +4373,7 @@
         },
         "string_decoder": {
           "version": "1.1.1",
-          "resolved": "https://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz",
+          "resolved": "http://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz",
```
Comment (Sw0rdstream) [inline package-lock.json]: This line is changed

File: migrate-mongo-store.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+}
+migrationSchema.statics.updateByEntry = function(entry) {
+  const { title } = entry
+  const at = entry.timestamp || Date.now()
```
Comment (Sw0rdstream) [inline migrate-mongo-store.js]: can also use destructing like this:
```
const { title, at = Date.now(), timestamp } = entry
```

File: migrate-mongo-store.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  const status = entry.timestamp === null ? 'down' : 'up'
```
Comment (Sw0rdstream) [inline migrate-mongo-store.js]: if timestamp is undefined or 0, status will be `up`. Is that your intention?

File: migrate-mongo-store.js (hunk 4 above)
Comment (otofune) [reply migrate-mongo-store.js]:
```js
const { title, timestamp: at = Date.now() } = entry
```

File: migrate-mongo-store.js (hunk 5 above)
Comment (otofune) [reply migrate-mongo-store.js]: I saw `tj/node-migrate`s implement: https://github.com/tj/node-migrate/blob/master/lib/migrate.js#L56-L57
I want to track 'down' event to know execution from another managers, but there are no hint without timestamp from store 😭

File: migrate-mongo-store.js (hunk 4 above)
Comment (Sw0rdstream) [reply migrate-mongo-store.js]: my bad

File: migrate-mongo-store.js (hunk 5 above)
Comment (otofune) [reply migrate-mongo-store.js]: I see, I'll use it! Thank you 🙏

**Pull request closed without merging**